i18n-helpers/src/lib.rs

```diff
@@ -367,14 +367,26 @@ pub fn group_events<'a>(events: &'a [(usize, Event<'a>)]) -> Vec<Group<'a>> {
 
                         ctx.comments.push(comment);
                     }
-                    // Otherwise, treat as a skipping group.
                     _ => {
-                        if let State::Translate(_) = state {
-                            let mut next_groups;
-                            (next_groups, ctx) = state.into_groups(idx, events, ctx);
-                            groups.append(&mut next_groups);
-
-                            state = State::Skip(idx);
+                        // Otherwise, treat as a skipping group if this is a block level Html tag
+                        if matches!(event, Event::Html(_)) {
```

Comment:
Could the `matches!` macro here be moved up into the `match` statement above?

That is, instead of a single match arm with

```rust
Event::Html(s) | Event::InlineHtml(s) => {
```

above, have separate `Event::Html(s)` and `Event::InlineHtml(s)` arms.

In general, the `matches!` macro is used rather rarely in Rust since the `match` or `if let` statements normally allow you to do the same thing.

Reply:
I was hesitant to duplicate the directives::find() call and handling of Skip and Comment.
I'll give it another pass

Reply:
Ah, I see! I didn't think of that :smile: 

I think changing the two `matches!` calls here to a nested `match` statement would be a small improvement. The whole logic here is becoming a little complex and it could very well be ripe for some refactoring (in a later PR).